worlds/openrct2/__init__.py

```diff
@@ -0,0 +1,526 @@
+import math
+from typing import TextIO
+from Utils import local_path
+
+import worlds.LauncherComponents as LauncherComponents
+from BaseClasses import ItemClassification, Region, Item, Location, Tutorial
+from worlds.generic.Rules import add_rule
+
+from .Constants import base_id, item_info, location_info, scenario_info
+from .Items import OpenRCT2Item, set_openRCT2_items
+from .Options import openRCT2Options, Scenario, openrct2_option_groups
+from worlds.AutoWorld import World, WebWorld
+
+import os
+import urllib.request
+
+class OpenRCT2WebWorld(WebWorld):
+    theme = "partyTime"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the OpenRCT2 randomizer on your computer.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Crazycolbster"]
+    )
+
+    tutorials = [setup_en]
+    option_groups = openrct2_option_groups
+
+
+class OpenRCT2Location(Location):
+    game = "OpenRCT2"
+
+
+def launch_client() -> None:  # Shoutout to Serpent.ai for the launcher code!
+    from .Client import main
+    LauncherComponents.launch_subprocess(main, name="OpenRCT2Client")
+
+
+LauncherComponents.components.append(
+    LauncherComponents.Component(
+        "OpenRCT2 Client",
+        func=launch_client,
+        component_type=LauncherComponents.Type.CLIENT,
+        # OpenRCT2 icon credit to the OpenRCT2 team: 
+        # https://github.com/OpenRCT2/OpenRCT2/blob/develop/resources/logo/icon_x96.png
+        icon='openrct2icon' 
+    )
+)
+
+
+LauncherComponents.icon_paths['openrct2icon'] = local_path('data', 'openrct2icon.png')
+
+def get_previous_region_from_OpenRCT2_location(location_number: int):
+    if location_number <= 2:
+        return "OpenRCT2_Level_0"
+    elif location_number == 3 or location_number == 4 or location_number == 5 or location_number == 6:
+        return "OpenRCT2_Level_1"
+    else:
+        divider = location_number - 6
+        region = math.ceil(divider / 8) + 1
+        return f"OpenRCT2_Level_{region}"
+
+
+class OpenRCT2World(World):
+    """
+    OpenRCT2 is a fan-made, open-source reimplementation of the classic simulation game. It faithfully preserves 
+    the original game while introducing modern improvements and expanded features. Players can construct intricate 
+    roller coasters, manage finances, and build the park of their dreams!
+    """
+
+    game = "OpenRCT2"
+    web = OpenRCT2WebWorld()
+
+    options_dataclass = openRCT2Options
+    options: openRCT2Options
+    topology_present = False  # show path to required location checks in spoiler
+    item_name_to_id = {name: id for id, name in enumerate(item_info["all_items"], base_id)}
+    location_name_to_id = {name: id for id, name in enumerate(location_info["all_locations"], base_id)}
+    item_name_groups = {
```

Comment:
You should probably make these more user-friendly, so "Roller Coasters" instead of "roller_coasters"

Reply:
It took some debugging, but I finally got it